src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.cs

```diff
@@ -57,6 +55,14 @@ public static partial class JsonSerializer
                         }
                         else
                         {
+                            ReadOnlySpan<byte> propertyName = reader.HasValueSequence ? reader.ValueSequence.ToArray() : reader.ValueSpan;
```

Comment:
Is there a way to leverage `TextEquals` to avoid having to do the `GetUnescapedString`/`ValueSequence.ToArray()` or is that not feasible since it would force you to pass along the reader to `JsonClassInfo.GetProperty`?

Reply:
Not feasible - the `GetProperty` call has it's own lookup and can't compare via `reader.TextEquals`.